lib/galaxy_test/api/test_datasets.py

```diff
@@ -86,6 +86,39 @@ def test_search_by_tool_id(self):
         result = self._get("datasets", payload).json()
         assert len(result) == 0
 
+    def test_search_by_extension(self):
+        self.dataset_populator.new_dataset(self.history_id, wait=True)
+        payload = {
+            "q": ["extension"],
+            "qv": ["txt"],
+            "history_id": self.history_id,
+        }
+        assert len(self._get("datasets", payload).json()) == 1
+        payload = {
+            "q": ["extension"],
+            "qv": ["bam"],
+            "history_id": self.history_id,
+        }
+        assert len(self._get("datasets", payload).json()) == 0
+        payload = {
+            "q": ["extension-in"],
+            "qv": ["bam,txt"],
```

Comment:
What's the advantage here of putting bam,txt into a list ?

Reply:
It's the same we do e.g. for filtering on `state`, e.g. https://usegalaxy.org/api/datasets?q=state-in&qv=error,ok

Not sure that answers your question?

Reply:
I guess it is just redundant, the list would be useful if we are adding multiple q/qv values to the request. But this depends on how the `requests` lib handles these params, which I'm not 100% familiar with.

Reply:
Alright, this is not great but I guess we're stuck now with this.